go/coinstacks/binance/api/handler.go

```diff
@@ -0,0 +1,246 @@
+
+		for _, m := range t.Messages {
+			if m.Addresses == nil {
+				continue
+			}
+
+			// unique set of addresses
+			for _, addr := range m.Addresses {
+				if _, ok := seen[addr]; !ok {
+					addrs = append(addrs, addr)
+					seen[addr] = true
+				}
+			}
+		}
+
+		return t, addrs, nil
+	})
+
+	err := h.WSClient.Start()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (h *Handler) StopWebsocket() {
+	h.WSClient.Stop()
+}
+
+func (h *Handler) GetInfo() (api.Info, error) {
+	info := Info{
+		Info: cosmos.Info{
+			BaseInfo: api.BaseInfo{
+				Network: "mainnet",
+			},
+		},
+	}
+
+	return info, nil
+}
+
+func (h *Handler) GetAccount(pubkey string) (api.Account, error) {
+	res := bnbsdk.BalanceAccount{}
+
+	_, err := h.HTTPClient.LCD.R().SetResult(&res).Get(fmt.Sprintf("/api/v1/account/%s", pubkey))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get account")
+	}
+
+	balances := map[string]string{"BNB": "0"}
+	for _, b := range res.Balances {
+		balances[b.Symbol] = strconv.Itoa(int(b.Free.ToInt64()))
+	}
+
+	a := Account{
+		Account: cosmos.Account{
+			BaseAccount: api.BaseAccount{
+				Balance:            balances["BNB"],
+				UnconfirmedBalance: "0",
+				Pubkey:             res.Address,
+			},
+			AccountNumber: int(res.Number),
+			Sequence:      int(res.Sequence),
+			Assets:        []cosmos.Value{},
+		},
+	}
+
+	return a, nil
+}
+
+func (h *Handler) GetTxHistory(pubkey string, cursor string, pageSize int) (api.TxHistory, error) {
+	//res := bnbsdk.BalanceAccount{}
+
+	//_, err := h.HTTPClient.LCD.R().SetResult(&res).Get(fmt.Sprintf("/api/v1/account/%s", pubkey))
+	//if err != nil {
+	//	return nil, errors.Wrap(err, "failed to get account")
+	//}
+
+	//txs := []Tx{}
+	//for _, t := range res.Txs {
+	//	tx, err := h.formatTx(t)
+	//	if err != nil {
+	//		return nil, errors.Wrapf(err, "failed to format transaction: %s", *t.TendermintTx.Hash)
+	//	}
+
+	//	txs = append(txs, *tx)
+	//}
+
+	//txHistory := TxHistory{
+	//	BaseTxHistory: api.BaseTxHistory{
+	//		Pagination: api.Pagination{
+	//			Cursor: res.Cursor,
+	//		},
+	//		Pubkey: pubkey,
+	//	},
+	//	Txs: txs,
+	//}
+
+	//return txHistory, nil
+	return nil, nil
+}
+
+func (h *Handler) GetTx(txid string) (api.Tx, error) {
+	var res *rpctypes.RPCResponse
+
+	_, err := h.HTTPClient.RPC.R().SetResult(&res).SetQueryParam("hash", txid).Get("/tx")
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get tx: %s", txid)
+	}
+
+	tx := &rpc.ResultTx{}
+	err = h.HTTPClient.GetEncoding().Amino.UnmarshalJSON(res.Result, tx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to decode tx: %v", res.Result)
+	}
+
+	pTx, err := rpc.ParseTx(h.HTTPClient.GetEncoding().Amino.Amino, tx.Tx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse tx: %v", tx.Tx)
+	}
+
+	block, err := h.BlockService.GetBlock(int(tx.Height))
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get block: %d", tx.Height)
+	}
+
+	t := Tx{
```

Comment:
TODO: need to get/calculate fees or hardcode if they are static and never change